aem-authoring-toolkit-assets/src/main/content/jcr_root/apps/authoring-toolkit/depends-on/js/dependsOnActionRegistry.js

```diff
@@ -38,12 +39,41 @@
             return action;
         }
 
+        /**
+         * Function to remove symbols that don't match to the pattern
+         * @param {string} target - target string
+         * @param {RegExp} pattern -  pattern for correct symbols
+         * */
+        static _replaceForbiddenSymbols(target, pattern) {
+           return target
+               .split('')
+               .map((symbol) => (symbol.search(pattern) !== -1  ? symbol : ''))
+               .join('');
+        }
+
+        /**
+         * Function to develop action's name (allowed symbols a-z, 0-9, -)
+         * @param {string} name - action name
+         * @returns {string} correct action name (according to nameRegex)
+         * */
+        static _refactorName(name) {
+            const nameRegex = /^[a-z0-9-]+$/;
+            let resultName = name;
+            if (name.trim().search(nameRegex) === -1) {
+                resultName = resultName.toLocaleLowerCase();
+                resultName = this._replaceForbiddenSymbols(resultName, nameRegex);
```

Comment:
maybe this method can be replaed with `resultName.replace(/[^a-z0-9-]+/g, '')` ?

Reply:
No, because we need to remove symbols that are **not** a-z0-9-

Reply:
The regex above does exactly this, e.g. `'AAAbbb#$#$$-121323'.replace(/[^a-z0-9-]+/g, '')` will return `bbb-121323` 
the `^` symbol after the opening bracket means that we are looking for symbols that are not a-z0-9-
But, it's up to you, perhaps your solution is more readable :)

Reply:
I definitely forgot about ^ symbol)